server/src/main/java/org/elasticsearch/action/datastreams/lifecycle/ExplainIndexDataStreamLifecycle.java

```diff
@@ -77,6 +81,11 @@ public ExplainIndexDataStreamLifecycle(
     public ExplainIndexDataStreamLifecycle(StreamInput in) throws IOException {
         this.index = in.readString();
         this.managedByLifecycle = in.readBoolean();
+        if (in.getTransportVersion().onOrAfter(TransportVersions.NO_LIFECYCLE_FOR_SYSTEM_DATA_STREAMS)) {
+            this.isSystemDataStream = in.readBoolean();
+        } else {
+            this.isSystemDataStream = false;
+        }
```

Comment:
Nit:
```suggestion
        this.isSystemDataStream = in.getTransportVersion().onOrAfter(TransportVersions.NO_LIFECYCLE_FOR_SYSTEM_DATA_STREAMS)) && in.readBoolean();
```

Reply:
I kind of like the simplicity of the if/else here. That's just a matter of personal preference, right? The outcome is the same either way?

Reply:
I'd argue that the one-liner is just as simple :stuck_out_tongue:. But it's definitely just personal preference as I've seen both variations and the outcome is the same. I'm fine with leaving it like this if that's your preference.